### PR TITLE
3670: Add put implementation timescales

### PR DESCRIPTION
**Pull request by kat1906**

### CODE REVIEW

File: tests/NHSD.BuyingCatalogue.API.IntegrationTests/Features/ImplementationTimescales/Suppliers Edit Solution Implementation Timescales Section.3670.feature [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            | SolutionID | SolutionName   | SupplierStatusId | SupplierId |
+            | Sln1       | MedicOnline    | 1                | Sup 1      |
+            | Sln2       | TakeTheRedPill | 1                | Sup 1      |
+            | Sln3       | PracticeMgr    | 1                | Sup 1      |
+        And SolutionDetail exist
+            | Solution | ImplementationTimescales                              |
+            | Sln1     | An original implementation timescales description      |
+            | Sln2     | Another original implementation timescales description |
+
+    @3667
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.API.IntegrationTests/Features/ImplementationTimescales/Suppliers Edit Solution Implementation Timescales Section.3670.feature]: Change this to @3670

File: tests/NHSD.BuyingCatalogue.API.IntegrationTests/Features/ImplementationTimescales/Suppliers Edit Solution Implementation Timescales Section.3670.feature [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    Scenario: 1. Solution implementation timescales section data is updated
+        When a PUT request is made to update the implementation-timescales section for solution Sln1
+            | ImplementationTimescales                    |
+            | A new implementation timescales description |
+        Then a successful response is returned
+        And SolutionDetail exist
+            | Solution | ImplementationTimescales                              |
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.API.IntegrationTests/Features/ImplementationTimescales/Suppliers Edit Solution Implementation Timescales Section.3670.feature]: Line endings not consistent

File: tests/NHSD.BuyingCatalogue.API.IntegrationTests/Features/ImplementationTimescales/Suppliers Edit Solution Implementation Timescales Section.3670.feature [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            | Sln1     | An original implementation timescales description      |
+            | Sln2     | Another original implementation timescales description |
+
+    @3670
+    Scenario: 1. Solution implementation timescales section data is updated
+        When a PUT request is made to update the implementation-timescales section for solution Sln1
+            | ImplementationTimescales                    |
+            | A new implementation timescales description |
+        Then a successful response is returned
+        And SolutionDetail exist
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.API.IntegrationTests/Features/ImplementationTimescales/Suppliers Edit Solution Implementation Timescales Section.3670.feature]: This needs to be completed. Go to where solution detail is and include this

Comment (kat1906) [reply tests/NHSD.BuyingCatalogue.API.IntegrationTests/Features/ImplementationTimescales/Suppliers Edit Solution Implementation Timescales Section.3670.feature]: I don't know what you mean by this

Comment (RuthBerry96) [reply tests/NHSD.BuyingCatalogue.API.IntegrationTests/Features/ImplementationTimescales/Suppliers Edit Solution Implementation Timescales Section.3670.feature]: This was sorted with the bug fix. Editing the actual step of solutionDetail

File: tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/TestContext.cs [hunk 4]
```diff
@@ -362,6 +370,8 @@ private class Scope
                 GetImplementationTimescalesBySolutionIdHandler = getImplementationTimescalesBySolutionIdHandler;
                 UpdateSupplierHandler = updateSupplierRequestHandler;
                 UpdateIntegrationsHandler = updateIntegrationsRequestHandler;
+                UpdateImplementationTimescalesHandler = updateImplementationTimescalesHandler;
+
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Solutions.Application.UnitTests/TestContext.cs]: Dont need this whitespace

File: tests/NHSD.BuyingCatalogue.Solutions.Persistence.DatabaseTests/SolutionDetailRepositoryTests.cs [hunk 5]
```diff
@@ -471,5 +471,80 @@ await SolutionEntityBuilder.Create()
                 .ConfigureAwait(false);
             result.Summary.Should().BeNull();
         }
+
+         [Test]
```
Comment (RuthBerry96) [inline tests/NHSD.BuyingCatalogue.Solutions.Persistence.DatabaseTests/SolutionDetailRepositoryTests.cs]: Can you make the [Test] text inline please :)

**Pull request merged**